### ISSUE TITLE
Pneumonia Detection using CNN(230112)

### ISSUE THREAD

**Issue by zzoall** [UGeunJi/Deep_Learning_Project#9]
🚑 pneumonia-detection
출처 : https://www.kaggle.com/code/madz2000/pneumonia-detection-using-cnn-92-6-accuracy

###  데이터 증강
![image](https://user-images.githubusercontent.com/117875637/212227792-e56700c4-6445-4a5f-92da-c96d58c24f2d.png)

### CNN 모델 생성
![image](https://user-images.githubusercontent.com/117875637/212229784-bfbe25aa-f7bb-4f21-aff5-c5435aff1265.png)

### 콜백함수
모델의 개선이 없을 경우, Learning Rate를 조절해 모델의 개선을 유도하는 케라스 콜백 함수
![image](https://user-images.githubusercontent.com/117875637/212229929-d89d4284-0079-4537-96f9-cd9720fc0172.png)

### 모델훈련
![image](https://user-images.githubusercontent.com/117875637/212230112-c5c2a9f0-c990-4048-a4e3-ef94183f8416.png)

### 결과
![image](https://user-images.githubusercontent.com/117875637/212230218-fc85fcdc-2e21-422b-bc6a-093fc72ac175.png)
->Train 학습 후 Test 예측에서 Loss는 0.29, Accuracy는 92% 를 볼 수 있다.

---
### 파라미터 수정했을 때의 결과가 궁금하여서 Fine-tuning을 시도했다.
---
[첫번째 시도]
1. 데이터 증강  vertical_flip을 False -> True
2. 마지막 Dropout을 0.2 -> 0.3 
3. 콜백함수 파라미터 patience = 2가 적은 것 같아서 5로 수정함

(train)
![image](https://user-images.githubusercontent.com/117875637/212230900-34634f7a-c774-4240-ae34-2575b352be8f.png)
(test)
![image](https://user-images.githubusercontent.com/117875637/212231092-9749c18d-65fc-446f-89b2-11657f2f8c82.png)

-> 학습의 결과는 95% 를 넘겼지만, 예측에서는 현저하게 낮아졌다. 많이 건들지 않다고 생각했는데 이렇게 낮아질 수가 있구나 ?_?

[두번째 시도]
1. vertical_flip = True
2. 마지막 Dropout 0.2
3. patience = 3
5. epoch 12 -> 25

(train)
![image](https://user-images.githubusercontent.com/117875637/212232073-d98bdfd2-d798-408d-9246-8046b6e7588e.png)
(test)
![image](https://user-images.githubusercontent.com/117875637/212232116-ea4aff80-86c4-4438-8784-c4175a40bc9a.png)

-> 에폭을 크게 늘린것이 성능향상에 도움이 된 것 같다.

[세번째 시도]
옵티마이저만 Adam으로 변경, 나머지는 동일하다.

(train)
![image](https://user-images.githubusercontent.com/117875637/212241318-25e0b77e-ec26-44c5-a55d-a00937cddf07.png)
(test)
![image](https://user-images.githubusercontent.com/117875637/212241367-73c3fae4-5ca9-4bde-8923-4a955602d385.png)

-> 정확도는 그대로인데 loss값이 0.5 정도 줄었다 ! Adam은 Adaptive Learning rate와 Momentum개념이 합쳐진 옵티마이저로 
손실함수 최적화를 위한 여러 방법 중에 자주 사용한다고 한다.
